work_queue/src/work_queue.c

```diff
@@ -6002,6 +6032,24 @@ int work_queue_shut_down_workers(struct work_queue *q, int n)
 	return i;
 }
 
+void work_queue_drain_worker(struct work_queue *q, const char *hostname)
```

Comment:
Since you added this function, we don't need work_queue_draininglist_*, nor q->worker_draininglist.

Add a draining_flag to work_queue_worker instead, and use this flag in  check_hand_against_task instead of hash_table_lookup(q->worker_draininglist, w->hostname) != NULL